Dynatrace/dynatrace-operator
# Fixes autoUpdate being enabled when applying a DynaKube in v1beta1 with autoUpdate set to false.

[K8S-10181](https://dt-rnd.atlassian.net/browse/K8S-10181)

Problem

Applying a DK v1beta1 with OneAgent autoUpdate set to false gets set to true by the conversion webhook.

The reason for this is how the object gets marshalled to JSON as autoUpdate is configured to be `omitempty`. as `false` is the default `bool` value it get's omitted and set to it's default value `true`.

## How can this be tested?

apply a DynaKube in version v1beta1 with 
```oneAgent:
    cloudNativeFullStack:
      autoUpdate: false
```

check the DK in both v1beta1 and v1beta2 and validate that autoUpdate is still false.
